protothread.h

```diff
@@ -427,6 +431,9 @@ pt_kill(pt_thread_t * const t)
     if (!pt_find_and_unlink(&s->ready, t)) {
         pt_thread_t ** const wq = pt_get_wait_list(s, t->channel) ;
         if (!pt_find_and_unlink(wq, t)) {
+            if (t->atexit) {
+                t->atexit(t->env) ;
+            }
```

Comment:
Why do you only want to run the `atexit` function if the thread is not on the ready list and not on the waiting list? Wouldn't you want to run it if the thread is found on either list? Maybe call `atexit` just before returning true?